### PR TITLE
wownero: integration

### PR DESCRIPTION
**Pull request by nahuhh**
- [ ] needs review & test 
- [x] js for wallet, wallets & offers templates @gerlofvanek 🙏 (todo post merge)
- [x] squash

~~this version has separate rpc. Seems using the rpc_xmr one might result in conflicting behavior~~

separate:
- [x] need to revert monero fee bump and update daemon #82

### CODE REVIEW

File: basicswap/interface/xmr.py [hunk 1]
```diff
@@ -81,9 +81,11 @@ def xmr_swap_b_lock_spend_tx_vsize() -> int:
     def __init__(self, coin_settings, network, swap_client=None):
         super().__init__(network)
 
+        self._addr_prefix = self.chainparams_network()['address_prefix']
+
         self.blocks_confirmed = coin_settings['blocks_confirmed']
         self._restore_height = coin_settings.get('restore_height', 0)
-        self.setFeePriority(coin_settings.get('fee_priority', 0))
+        self.setFeePriority(coin_settings.get('fee_priority', 2))
```
Comment (nahuhh) [inline basicswap/interface/xmr.py:88]:
```suggestion
        self.setFeePriority(coin_settings.get('fee_priority', 0))
```

File: basicswap/templates/settings.html [hunk 2]
```diff
@@ -195,10 +195,10 @@ <h4 class="font-semibold text-black dark:text-white text-2xl align-middle">
...
                  </svg>
                  <select class="hover:border-blue-500 bg-gray-50 text-gray-900 appearance-none pr-10 dark:bg-gray-500 dark:text-white border border-gray-300 dark:border-gray-400 dark:text-gray-50 dark:placeholder-gray-400 text-sm rounded-lg outline-none focus:ring-blue-500 focus:border-blue-500 block w-full p-2.5 focus:ring-0" name="fee_priority_{{ c.name }}">
-                  <option value="0" {% if c.fee_priority==0 %} selected{% endif %}>Default</option>
-                  <option value="1" {% if c.fee_priority==1 %} selected{% endif %}>Low</option>
-                  <option value="2" {% if c.fee_priority==2 %} selected{% endif %}>Medium</option>
-                  <option value="3" {% if c.fee_priority==3 %} selected{% endif %}>High</option>
+                  <option value="0" {% if c.fee_priority==0 %} selected{% endif %}>[Auto] Slow</option>
```
Comment (nahuhh) [inline basicswap/templates/settings.html:198]:
```suggestion
                  <option value="0" {% if c.fee_priority==0 %} selected{% endif %}>Auto</option>
```

File: bin/basicswap_prepare.py [hunk 3]
```diff
@@ -686,18 +743,18 @@ def prepareCore(coin, version_data, settings, data_dir, extra_opts={}):
             assert_sig_path = os.path.join(bin_dir, assert_sig_filename)
             if not os.path.exists(assert_sig_path):
                 downloadFile(assert_sig_url, assert_sig_path)
+    if coin != 'wownero':
```
Comment (nahuhh) [inline bin/basicswap_prepare.py:746]: Need to revert after hashes available, or a different check

File: bin/basicswap_prepare.py [hunk 4]
```diff
@@ -828,17 +888,25 @@ def prepareDataDir(coin, settings, chain, particl_mnemonic, extra_opts={}):
...
 
+    if coin in ('wownero', 'monero'):
         wallets_dir = core_settings.get('walletsdir', data_dir)
         if not os.path.exists(wallets_dir):
             os.makedirs(wallets_dir)
 
-        wallet_conf_path = os.path.join(wallets_dir, coin + '_wallet.conf')
+        wallet_conf_path = os.path.join(wallets_dir, coin + '-wallet-rpc.conf')
+        if coin == 'monero':
+            wallet_conf_path = os.path.join(wallets_dir, 'monero_wallet.conf')
```
Comment (nahuhh) [inline bin/basicswap_prepare.py:909]: Left monero_wallet.conf file alone, but renamed future integrations

File: bin/basicswap_prepare.py [hunk 5]
```diff
@@ -901,7 +971,7 @@ def prepareDataDir(coin, settings, chain, particl_mnemonic, extra_opts={}):
             fp.write('zmqpubsmsg=tcp://{}:{}\n'.format(COINS_RPCBIND_IP, settings['zmqport']))
             fp.write('spentindex=1\n')
             fp.write('txindex=1\n')
-            fp.write('staking=0\n')
+            fp.write('staking=1\n')
```
Comment (nahuhh) [inline bin/basicswap_prepare.py:974]: staking was disabled

File: bin/basicswap_prepare.py [hunk 6]
```diff
@@ -1200,12 +1274,17 @@ def initialise_wallets(particl_wallet_mnemonic, with_coins, data_dir, settings,
...
+                        filename = coin_name + '-wallet-rpc' + ('.exe' if os.name == 'nt' else '')
+                        daemons.append(startXmrWalletDaemon(coin_settings['datadir'], coin_settings['bindir'], filename))
+                elif c == Coins.WOW:
+                    if coin_settings['manage_wallet_daemon']:
+                        filename = coin_name + '-wallet-rpc' + ('.exe' if os.name == 'nt' else '')
                         daemons.append(startXmrWalletDaemon(coin_settings['datadir'], coin_settings['bindir'], filename))
                 else:
                     if coin_settings['manage_daemon']:
                         filename = coin_name + 'd' + ('.exe' if os.name == 'nt' else '')
-                        coin_args = ['-nofindpeers', '-nostaking'] if c == Coins.PART else []
```
Comment (nahuhh) [inline bin/basicswap_prepare.py:1208]: Hm

**Review by nahuhh — COMMENTED** [review]
Left some comments / questions

File: bin/basicswap_prepare.py (hunk 5 above)
Comment (tecnovert) [reply bin/basicswap_prepare.py:974]: 1. Staking is disabled on purpose.
2. What has this got to do with wownero?

Comment (nahuhh) [reply bin/basicswap_prepare.py:974]: Ill revert these

File: basicswap/interface/wow.py [hunk 7]
```diff
@@ -0,0 +1,574 @@
...
+    def coin_type():
+        return Coins.WOW
+
+    @staticmethod
+    def COIN():
+        return WOW_COIN
+
+    @staticmethod
+    def exp() -> int:
+        return 11
```
Comment (nahuhh) [inline basicswap/interface/wow.py:58]: @tecnovert is this line for atomic units (decimal places?)
XMR was 12, wownero is 11. Changed to 11. 

please let me know. 
thanks

**Comment by nahuhh** [comment]
rebased for decred changes using another branch https://github.com/nahuhh/basicswap/compare/wownero_full...nahuhh:basicswap:rbf_wownero

**Pull request closed without merging**